src/components/TextInput/native/TextInput.js

```diff
@@ -0,0 +1,233 @@
+import React, { Component } from 'react';
+import { Animated } from 'react-native';
+import PropTypes from 'prop-types';
+import styled, { withTheme } from 'styled-components/native';
+import Text from '../../Text/native';
+
+const modifierColor = (focused = 'green', valid = 'greyLight', invalid = 'red', disabled = 'grey') => (props) => {
+  if (props.isFocused) return props.theme.color[focused];
+  if (props.disabled) return props.theme.color[disabled];
+  return props.error ? props.theme.color[invalid] : props.theme.color[valid];
+};
+
+const labelColor = (props) => {
```

Comment:
Consult with the design team to see if this use case could be better handled within `modifierColor` itself